packages/patternfly-4/react-core/src/components/Dropdown/DropdownMenu.js

```diff
@@ -27,25 +27,85 @@ const defaultProps = {
   component: 'ul'
 };
 
-const DropdownMenu = ({ className, isOpen, position, children, component: Component, ...props }) => {
-  let menu = null;
-  if (Component === 'div') {
-    menu = (
-      <Component
-        {...props}
-        className={css(
-          styles.dropdownMenu,
-          position === DropdownPosition.right && styles.modifiers.alignRight,
-          className
-        )}
-        hidden={!isOpen}
-      >
-        {children}
-      </Component>
+class DropdownMenu extends React.Component {
+  constructor(props) {
+    super(props);
+    this.refsCollection = {};
+    this.keyHandler = this.keyHandler.bind(this);
+  }
+
+  keyHandler(index, position, custom = false) {
+    const kids = this.props.children;
```

Comment:
This could break if consumers put items in a fragment. It might be better to make use of context to pass things around.

Reply:
I'm not sure what you mean by using context? I am seeing that having a child that is a React.Fragment will throw an error but should we just handle that when the children get extended (wrapping them in a div) or is there a better method?

Reply:
The context API https://reactjs.org/docs/context.html

Reply:
I can move keyHandler and sendRef to a Context but they only get passed down one layer so I'm not sure it saves much. I'm also confused about the case of putting the items in a Fragment because the top level Dropdown component should be expecting an array.